.github/workflows/validate-with-registry.yaml

```diff
@@ -0,0 +1,66 @@
+name: Validate with Devfile Registry Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  validate-devfile:
+    runs-on: ubuntu-latest
+
+    env:
+      REGISTRY_PATH: ${{ github.workspace }}/registry
+
+    steps:
+    - name: Checkout current repo
+      uses: actions/checkout@v4
+      with:
+        path: current-repo
+
+    - name: Checkout devfile registry
+      uses: actions/checkout@v4
+      with:
+        repository: devfile/registry
+        path: registry
+
+    - name: Setup test environment with overrides
+      run: |
+        mkdir -p registry/stacks/python-basic
+        cp -r current-repo/* registry/stacks/python-basic/
+
+    - name: Setup Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.21'
+
+    - name: Install Ginkgo
+      run: |
+        sudo wget -qO /usr/local/bin/yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64
+        sudo chmod +x /usr/local/bin/yq
+        go install github.com/onsi/ginkgo/v2/ginkgo@latest
+
+    - name: Setup Minikube
+      uses: medyagh/setup-minikube@master
+      with:
+        minikube-version: latest
+        kubernetes-version: v1.28.0
+
+    - name: Run Registry Validation Tests
+      env:
+        REGISTRY_PATH: ${{ github.workspace }}/registry
+        TEST_NAMESPACE: "default"
+        YQ_PATH: "yq"
+        ENV: "minikube"
+      run: |
+        cd registry
+       
+        echo "=== Schema Validation ==="
+        bash tests/validate_devfile_schemas.sh --stacksPath stacks --stackDirs python-basic
```

Comment:
Here we need to add the ``--samples`` flag right?

Based on the code here: https://github.com/devfile/registry/blob/main/tests/validate_devfile_schemas.sh#L53-L63